### ISSUE TITLE
Rhinoceros3d crashes at certain points; SBIE2101

### ISSUE THREAD

**Issue by chytheche7** [sandboxie-plus/Sandboxie#1448]
**Describe the problem**
When invoking certain functions within the 3d modeling application Rhinoceros3d ("Rhino"), the program crashes abruptly, with Sandboxie giving the Message:
`WerFault.exe (5552): SBIE2101 Object name not found: Unnamed object, error OpenProcess (C0000022) access=001FFFFF initialized=1`

As an example I describe the crash as it happens when trying to "Save layer states" within Rhino. Other functions that cause this sort of crash are: Saving a "Named View", saving a "Snapshot" of the model's current configuration, applying "Box Edit". There are probably many other functions causing this crash. These are a few I identified in a short time.

**To Reproduce**
1) You enable the "Layer States" tab within the toolbar to the right by pressing the gearwheel symbol and selecting "Layer States" from the dropdown menu:
![1InkedAnnotation 2021-12-18 105556](https://user-images.githubusercontent.com/96419076/146894549-586e3fc3-c788-449c-84fe-47f2b805e189.jpg)
2) You navigate to the tab "Layer States":
![2InkedAnnotation 2021-12-18 105657](https://user-images.githubusercontent.com/96419076/146894649-26fd12f6-faa0-4372-85c8-70803272b6e5.jpg)
3) The crash happens when you click the "save" icon. It happens instantaneously and no further changes within Rhino are visible.
![3InkedAnnotation 2021-12-18 105716](https://user-images.githubusercontent.com/96419076/146894808-8a9420a6-e596-4067-8418-7d45a33ba4d5.jpg)

**Expected behavior**
A new item should appear in the table below, starting off with name editing active and the default name being highlighted.
![supposedtohappen_Inked9OqHPddL0T](https://user-images.githubusercontent.com/96419076/146894992-1a4a53a5-4bd4-44b7-8c79-d463d8640f69.jpg)

**System details and installed software**
Clean Windows 10 guest VM on VirtualBox 6.1.30 r148432 (Qt5.6.2)
Windows 10 Education
10.0.19043 Build 19043
Intel Core i5-8265U (8th generation)

Installed Software: Sandboxie Plus x64 1.0.3
Rhinoceros 3d 7.13.21348.13001

**If you had a compatibility issue or a crash**
Prerequisite: Install Rhino 7.13.21348.13001 directly onto the system (meaning not sandboxed).
1) Download the installer: https://files.mcneel.com/dujour/exe/20211214/rhino_en-us_7.13.21348.13001.exe (trial installer).
2) Within the installer, change the license method to "This computer":
![1InkedAnnotation 2021-12-18 114423](https://user-images.githubusercontent.com/96419076/146896470-c2da6d7e-60a2-4a47-b269-7703a8564f87.jpg)
![2Inked2InkedAnnotation 2021-12-18 113136](https://user-images.githubusercontent.com/96419076/146896501-cf54ac20-bb59-476f-a5b8-e65eed7a6247.jpg)
3) Enter the license key RH70-RQ00-LG04-9H0U-8HW4-109Q (anonymous trial license); Press "Install now":
![3Inked3Inked2Annotation 2021-12-18 114825](https://user-images.githubusercontent.com/96419076/146896520-27cdc385-1af0-491e-93db-224676aae01c.jpg)
4) On program start (sandboxed), click away the request for entering an email address: 
![4InkedAnnotation 2021-12-18 115228](https://user-images.githubusercontent.com/96419076/146896735-c9eba3aa-2c91-4992-9979-eb177a89ed5d.jpg)
5) A popup should appear. Press "Ask me later":
![5InkedAnnotation 2021-12-18 115240](https://user-images.githubusercontent.com/96419076/146896756-3e6f57c3-73b0-44cc-8e86-df146e448aaa.jpg)

**Sandboxie traces and dmp file**
Sandboxie log with basic logging options: [basic log hosted on gist](https://gist.github.com/chytheche7/ade2bf6bc5e4264d8de6890633e2f8ef)
<details>

<summary>sandboxie.ini basic logging</summary>

```

IpcTrace=ad
PipeTrace=ad
GuiTrace=ad

```

</details>

Sandboxie log with "complete" logging options: [extended log file hosted on github ~40MB](https://github.com/chytheche7/Filehost_Rhinoceros3d-SBIE2101-Debug-Logs/blob/69786ed3097fb5a45d4b463278692c7cdde4245b/211221%20Rhino%20SBIE2101%20trace%20log%20extended.txt)
<details>

<summary>sandboxie.ini "complete" logging</summary>

```

FileTrace=ad
KeyTrace=ad
PipeTrace=ad
IpcTrace=ad
GuiTrace=ad
ClsidTrace=ad
NetFwTrace=ad
LogAPI=ad

```

</details>

Dmp file as created by Rhino:  [dmp file hosted on github](https://github.com/chytheche7/Filehost_Rhinoceros3d-SBIE2101-Debug-Logs/blob/69786ed3097fb5a45d4b463278692c7cdde4245b/211221%20Rhino%20SBIE2101%20Rhino.exe.4132.dmp)
<details>

<summary>Guide on reading Rhino dmp</summary>

```

    Open Visual Studio
        From the Tools menu, click Options
        Select the Debugging > Symbols tab
        In the Symbol file (.pdb) locations box, add:
            http://s3.symbols.rhino3d.com/symbols/dujour
        Optionally add these symbol servers if you need symbols for OpenGL driver related crashes:
            https://download.amd.com/dir/bin_2018
            https://download.amd.com/dir/bin
            https://driver-symbols.nvidia.com

```

</details>

**Additional context**
1) The application is installed outside the Sandbox directly onto the system in order to avoid potential further complications with dotnet and Visual C++ redistributables.

2) After experiencing the crash on the host machine, I switched to a VM guest OS for the purpose of trying different Sandboxie settings and combinations thereof. The crash happens just the same in both setups.

3) The only Sandboxie configuration I found which prevented the crash from happening is the following with minimal isolation, which, to my understanding, is similar to not running sandboxed at all:
<details>

<summary>sandboxie.ini minimal isolation</summary>

```

BoxNameTitle=-  
NoAddProcessToJob=y  
NoSandboxieDesktop=y  
OriginalToken=y  
NoSysCallHooks=y  
OpenIpcPath=*  
OpenWinClass=*

```

</details>

4) Also, the "Box Type" presets recently introduced to Sandboxie ("Application compartment" and so forth) could not prevent this crash.

**Sandboxie configuration**
The Sandboxie configuration used is the default one. 

<details>

<summary>sandboxie.ini</summary>

```

[GlobalSettings]
FileRootPath=\??\%SystemDrive%\Sandbox\%USER%\%SANDBOX%
SeparateUserFolders=y
KeyRootPath=\REGISTRY\USER\Sandbox_%USER%_%SANDBOX%
IpcRootPath=\Sandbox\%USER%\%SANDBOX%\Session_%SESSION%
NetworkEnableWFP=n
EnableObjectFiltering=n
EnableWin32kHooks=n
EditAdminOnly=n
ForceDisableAdminOnly=n
ForgetPassword=n
Template=WindowsRasMan
Template=WindowsLive
Template=OfficeLicensing

[UserSettings_0C9801F1]
SbieCtrl_AutoStartAgent=SandMan.exe
SbieCtrl_EnableAutoStart=y

[DefaultBox]
Enabled=y
AutoRecover=n
BlockNetworkFiles=y
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Desktop%
BorderColor=#00FFFF,ttl
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=Chrome_Phishing_DirectAccess
Template=Firefox_Phishing_DirectAccess
Template=AutoRecoverIgnore
ConfigLevel=9

```

</details>

**PS**
Edit 21-12-23: Changed filehost and access links for larger (~40MB) log files.

**Comment by isaak654** [comment]
> Sandboxie log with "complete" logging options: https://we.tl/t-Lt8RLtAs6I (too large for gist)
> sandboxie.ini "complete" logging
> 
> Dmp file as created by Rhino: https://we.tl/t-G2px1JYmMi

I would upload them into another file hosting service, because these links are set to expire in 6 days.
The maintainer can't assure that he will take a look at them before expiration.

**Comment by isaak654** [comment]
> Sandboxie log with "complete" logging options: [extended log file hosted on github ~40MB](https://github.com/chytheche7/Filehost_Rhinoceros3d-SBIE2101-Debug-Logs/blob/69786ed3097fb5a45d4b463278692c7cdde4245b/211221%20Rhino%20SBIE2101%20trace%20log%20extended.txt)
> sandboxie.ini "complete" logging
> 
> Dmp file as created by Rhino: [dmp file hosted on github](https://github.com/chytheche7/Filehost_Rhinoceros3d-SBIE2101-Debug-Logs/blob/69786ed3097fb5a45d4b463278692c7cdde4245b/211221%20Rhino%20SBIE2101%20Rhino.exe.4132.dmp)

The new links you uploaded return "Page not found" errors when I try to open them...

**Comment by DavidXanatos** [comment]
I have downloaded the dump and teh log file

**Comment by DavidXanatos** [comment]
I think i found the issue it will be fixed in the next build 1.0.5
thank you very much for the great and detailed bug report.

**Comment by chytheche7** [comment]
Great!

Thank your isaak654 for checking the links again. I had the repo at the target set to "private". It should be fixed now.
Thank you, davidxanatos and everyone involved, very much for your magnificent work on sandboxie plus. Writing the issue report with some thoroughness was the least I could do - I'm glad it was satisfactory. Keep it up!
